README.md

```diff
@@ -1,352 +1,372 @@
+[`az containerapp up`](https://docs.microsoft.com/en-us/cli/azure/containerapp?view=azure-cli-latest#az-containerapp-up)
+command. If `azure/login` is not called prior to this action being called in the GitHub workflow, the `azureCredentials`
+argument can be used to provide the credentials needed to successfully call `azure/login`. These credentials are
+recommended to be stored as a [GitHub secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets),
+and can be provided to this action without their value being exposed.
+
+The credentials used for the `azure/login` action must have Contributor access over one of the following sets of
+permissions:
+
+- The existing Container App if both the `resourceGroup` and `containerAppName` arguments are provided _and_ exist
+- The existing resource group if only the `resourceGroup` argument is provided _and_ exists
+- The user's subscription if the `resourceGroup` argument is not provided _or_ is provided but does not exist
+
+More information about configuring the deployment credentials required for this GitHub Action can be found
+[here](https://github.com/marketplace/actions/azure-login#configure-deployment-credentials).
+
+### `docker/login-action`
+
+The [`docker/login-action`](https://github.com/marketplace/actions/docker-login) action is used to authenticate calls
+to the user's Azure Container Registry, which will host the image that is then deployed to the Container App.
+Currently, `docker/login-action` is called during every invocation of this action, so the user's Azure Container
+Registry name is required, along with username and password credentials that are able to authenticate calls to this
+Azure Container Registry. These credentials are able to be retrieved by
+[creating a service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal#create-a-service-principal)
+and giving it proper permissions to the ACR resource.
+
+### pack CLI
+
+The [pack CLI](https://buildpacks.io/docs/tools/pack/) is maintained by the Cloud Native Buildpacks project and is used
+by this action to create runnable application images for the user when the application source code is provided and no
+additional Dockerfile is provided or found. A [builder](https://buildpacks.io/docs/concepts/components/builder/) was
+created by Oryx to take in the application source code provided to this action and produce an image that could then be
+pushed to an image registry and used within a Container App to build and run the application.
+
+A stable version of the pack CLI is installed on the GitHub runner executing the task, and depending on the base OS of
+this runner, different tools will be leverage to assist with the installation:
+- On Windows runners:
+  - `curl` will be used to pull down the `.zip` containing the `pack` executable
+  - `7z` will be used to unzip the `.zip` and place the `pack` executable in a new folder

```

Comment:
Nit: would be nice to state where `pack` is placed, in Windows. Since we let specify where `pack` is placed for Linux

Reply:
Or alternately not mentioning it for Linux? I'm not sure that's super valuable for customers. Do you see cases where they need to know that?

Reply:
@daniv-msft it may be "nice-to-know" for the customer, so I just extended this sentence to call out that the executable was placed in a "pack" folder that's in the action's working directory